fuse/utils/data/collate.py

```diff
@@ -124,49 +113,50 @@ def uncollate(batch: Dict) -> List[Dict]:
     Reverse collate method
     Gets a batch_dict and convert it back to list of samples
     """
-    samples = []
-    if not isinstance(batch, NDict):
-        batch = NDict(batch)
-    keys = batch.keypaths()
-
     # empty batch
-    if not keys:
-        return samples
+    if not batch.keys():
+        return []
 
-    if "data.sample_id" in keys:
+    if isinstance(batch, NDict):
+        batch = batch.flatten()
+
+    # infer batch size
+    if "data.sample_id" in batch:
         batch_size = len(batch["data.sample_id"])
     else:
         batch_size = None
 
-    if batch_size is None:
+        keys = batch.keys()
+
         for key in keys:
             if isinstance(batch[key], torch.Tensor):
```

Comment:
Why there are two different loops? one for each case - `torch.Tensor`, `(np.ndarray, list)`
If I'm not missing something we can check for the two cases in the same loop.

Reply:
Because, first I want to look for tensor (trust it more), and if I can't find one then my second choice is (np.ndarray, list)